docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+
+Because many WordPress themes place widgets into a sidebar column, widget areas are often referred to as _sidebars_ or _dynamic sidebars_.
+
+**For more terminology and a detailed overview of how widgets _currently_ work in WordPress, see [#14182][14182].**
+
+[14182]: https://github.com/WordPress/gutenberg/issues/14182
+
+## Prior reading
+
+- [#14182][14182] is the issue which tracks adding an API for reading and updating blocks in widget areas.
+- [#13204][13204] contains design discussion about showing blocks in `wp-admin/widgets.php`.
+- [#13205][13205] contains design discussion about showing blocks in the Customizer.
+- [#13511][13511] is a PR which added support for rendering existing widgets as a block in Gutenberg.
+
+[13204]: https://github.com/WordPress/gutenberg/issues/13204
+[13205]: https://github.com/WordPress/gutenberg/issues/13205
+[13511]: https://github.com/WordPress/gutenberg/issues/13511
+
+## Frontend
+
+A new `@wordpress/edit-widgets` package contains the UI for the `wp-admin/widgets.php` screen. This is analogous to how `@wordpress/edit-post` contains the UI for `wp-admin/post-new.php`. 
+
+Both WP Admin and the Customizer use the existing [`BlockEditor`][block-editor] component, which provides a generic UI for working with blocks, to display an interface for working with blocks in widget-editing areas.
+
+A new REST API, described in the next section, allows fetching and updating the blocks in a widget area. `@wordpress/edit-widgets` uses this API to fetch blocks which are passed to `BlockEditor`, and to update blocks when `BlockEditor` indicates that changes have been made by the user.
+
+```js
+function WidgetArea( { id } ) {
+	return (
+		<BlockEditor
+			// Illustrative only: apiGet() is not a real function. In production,
+			// @wordpress/core-data would be used to interface with the REST API.
+			blocks={ apiGet( `/wp/v2/widget-areas/${ id }` ) }
+			onChange={ ( blocks ) => {
+				// Illustrative only: apiPut() is not a real function. In production,
+				// @wordpress/core-data would be used to interface with the REST API.
+				apiPut( `/wp/v2/widget-areas/${ id }`, blocks );
+			} }
+		/>
+	);
+}
+```
+
+[block-editor]: https://github.com/WordPress/gutenberg/tree/master/packages/block-editor
+
+## REST API
+
+Fetching and updating the blocks in a widget area is done via new REST API endpoints.
+
+### The widget area resource
+
+The REST API is built around a single resource object: the widget area.
+
+```json
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "
+			<!-- wp:paragraph -->
+			<p>Hello there!</p>
+			<!-- /wp:paragraph -->
+
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+		",
+		"rendered": "
+			<p>Hello there!</p>
+			<form role=\"search\" method=\"get\" class=\"search-form\" action=\"http://wordpress.test/\">
+				...
+			</form>
+		",
+	},
+}
+```
+
+#### `id`
```

Comment:
Should we have a reserved "id"/"name" for the widgets that are kept separate when you switch themes and don't belong to any area defined by the theme?

Reply:
Good idea. I say we go with `wp_inactive` to align with what Core currently uses which is `wp_inactive_widgets`. Added in 3d698ffa9.